how-it-works/7_smart_contracts/01_canister-lifecycle.card.md

```diff
@@ -0,0 +1,15 @@
+---
+title: Canisters
+---
+
+![](/img/how-it-works/canister-lifecycle.600x300.jpg)
+
+# Canisters
+
+On the Internet Computer smart contracts come in the form of canisters.  These are computational units which bundle together code and state. Each canister defines a set of entry points which can be called by other canisters and parties external to the IC, such as browsers or mobile apps.
+When deployed, canisters behave much like actors from the actor-based concurrency model – they communicate asynchronously with each other but, unlike in the traditional actor model, communication is bidirectional. 
+Canisters are managed by controllers. These can deploy the canister to the Internet Computer, start/stop their execution and update their code.  As they execute, canisters need to acquire resources for their execution (memory, network bandwidth and computational power). To this end canisters hold, locally, a cycle balance which is the unit used to measure resource consumption. The controllers also need to ensure that canisters hold sufficient cycles. 
```

Comment:
Overall, this seems good. I was just thinking whether it makes sense to more immediately say that controllers can be any of (a) centralized, (b) empty – which makes the canister immutable – or (c) governed by a DAO. I think the notion of a *controller* may evoke images of centralized control, which is possible but not necessary. Anyway, just my reading and I may be wrong.

Reply:
Implemented